ClientApp/viking-vault/src/AdminDashboard/AddMoneyModal.tsx

```diff
@@ -35,10 +35,26 @@ class AddMoneyModal extends React.Component<IAddMoneyModalProps, IAddMoneyModalS
         this.props.addMoney(Number(this.state.amount));
     }
 
-    private handleAmountChange = (e:any) => this.setState({ amount: e.target.value });
+    private checkAmountValid = () =>{
+        if(!this.validateAmount()){
+            return  {
+                message: "Amount should be between 0.1-5000 RON",
+                class: "alert alert-danger"
+            }
+        }
+        return{
+            message: "",
+            class: ""
+        }
+    }
+
+    private handleAmountChange = (e:any) => {
+        this.setState({ amount: e.target.value 
+        }, this.checkAmountValid);
+    }
 
     private validateAmount(){
-        if(Number(this.state.amount) < this.state.minAmount || Number(this.state.amount) > this.state.maxAmount || !(this.state.amount.match("^[-+]?[0-9]*\.?[0-9]+([eE][-+]?[0-9]+)?$"))){
+        if(Number(this.state.amount) < this.state.minAmount || Number(this.state.amount) > this.state.maxAmount || !(this.state.amount.match("^[-+]?[0-9]*\.?[0-9]+([-+]?[0-9]+)?$"))){
```

Comment:
At this point it would be better to just separate each of the items in the `||` chain into their own variables and use them